src/SOLID/SingleResponsabilityPrinciple/Trial/BranchOffice.php

```diff
@@ -0,0 +1,48 @@
+<?php
+
+namespace Deg540\PHPTestingBoilerplate\SOLID\SingleResponsabilityPrinciple\Trial;
+
+class BranchOffice
+{
+    /**
+     * @var Employee[]
+     */
+    private array $employees;
+
+    public function insertEmployee(Employee $employee): void
+    {
+        $this->employees[] = $employee;
+    }
+
+    public function removeEmployee(Employee $employee): void
+    {
+        foreach ($this->employees as $key => $employeeInArray) {
+            if ($employeeInArray->getName() === $employee->getName()) {
+                unset($this->employees[$key]);
+            }
+        }
+    }
+
+    public function getEmployeesData(): string
+    {
+        $employeeData = '';
```

Comment:
Cuando me revisó David algo de fieldeas hizo incapie en utilizar las funciones de array_*** para realizar los bucles. 
¿qué te parece esta solución?

```suggestion
        $data = array_map(function ($employee) {
            return $employee->getData();
        }, $this->employees);

        return implode(' || ', $data);
```

Reply:
Hablando mal y pronto "cojonuda"

Reply:
Cambiado